lib/ansible/galaxy/role.py

```diff
@@ -438,7 +438,7 @@ def install(self):
                 if not (self.src and os.path.isfile(self.src)):
                     try:
                         os.unlink(tmp_file)
-                    except (OSError, IOError) as e:
+                    except OSError as e:
                         display.warning(u"Unable to remove tmp file (%s): %s" % (tmp_file, to_text(e)))
```

Comment:
When turning controller-side errors into warnings, use `display.error_as_warning`:

```suggestion
                        display.error_as_warning(f"Unable to remove tmp file {tmp_file!r}.", exception=ex)
```

We don't have an equivalent function in module_utils yet, but I expect we'll add one soon.